packages/docsearch-react/src/DocSearchModal.tsx

```diff
@@ -285,16 +303,23 @@ export function DocSearchModal({
                         .map(transformItems)
                         .map((groupedHits) =>
                           groupedHits.map((item) => {
+                            let parent: InternalDocSearchHit | null = null;
+
+                            const potentialParent = groupedHits.find(
+                              (siblingItem) =>
+                                siblingItem.type === 'lvl1' &&
+                                siblingItem.hierarchy.lvl1 ===
+                                  item.hierarchy.lvl1
+                            ) as InternalDocSearchHit;
+
+                            if (item.type !== 'lvl1' && potentialParent) {
+                              parent = potentialParent;
+                            }
```

Comment:
What's this change doing? How is it related to Insights?

Reply:
This was more of a cleanup to help with the typings of things. Overall no logic changes were made, just working around TypeScript